pkg/db/drivers/etcd/etcd.go

```diff
@@ -190,6 +190,184 @@ func (c *Client) ParameterFilter(m map[string][]string, size int, sortKeys []str
+        case "POOLID":
+                return fileshare[i].PoolId < fileshare[j].PoolId
+        case "DESCRIPTION":
+                return fileshare[i].Description < fileshare[j].Description
+        }
+        return false
+}
+
+func (c *Client) FindFileShareValue(k string, p *model.FileShareSpec) string {
+        switch k {
+        case "Id":
+                return p.Id
+        case "CreatedAt":
+                return p.CreatedAt
+        case "UpdatedAt":
+                return p.UpdatedAt
+        case "TenantId":
+                return p.TenantId
+        case "UserId":
+                return p.UserId
+        case "Name":
+                return p.Name
+        case "Description":
+                return p.Description
+        case "AvailabilityZone":
+                return p.AvailabilityZone
+        case "Size":
+                return strconv.FormatInt(p.Size, 10)
+        case "Status":
+                return p.Status
+        case "PoolId":
+                return p.PoolId
+        case "ProfileId":
+                return p.ProfileId
+        }
+        return ""
+}
+
+func (c *Client) CreateFileShare(ctx *c.Context, fshare *model.FileShareSpec) (*model.FileShareSpec, error) {
+        fmt.Println("inside the /driver/etct.go CreateFileShare(ctx")
+	//profiles, err := c.ListProfiles(ctx)
+	//if err != nil {
+	//	return nil, err
+	//} else if len(profiles) == 0 {
+	//	return nil, errors.New("No profile in db.")
+	//}
+
+	fshare.TenantId = ctx.TenantId
+	fshareBody, err := json.Marshal(fshare)
+	if err != nil {
+		return nil, err
+	}
+
+	dbReq := &Request{
+		Url:     urls.GenerateFileShareURL(urls.Etcd, ctx.TenantId, fshare.Id),
+		Content: string(fshareBody),
+	}
+	dbRes := c.Create(dbReq)
+        fmt.Println("inside the /driver/etct.go dbRes:= ",dbRes)
+	if dbRes.Status != "Success" {
+		log.Error("When create fileshare in db:", dbRes.Error)
+		return nil, errors.New(dbRes.Error)
+	}
+
+	return fshare, nil
+}
+
+func (c *Client) SelectFileShares(m map[string][]string, fileshares []*model.FileShareSpec) []*model.FileShareSpec {
+        if !c.SelectOrNot(m) {
+                return fileshares
+        }
+
+        var vols = []*model.FileShareSpec{}
+        var flag bool
+        for _, vol := range fileshares {
+                flag = true
+                for key := range m {
+                        if utils.Contained(key, validKey) {
+                                continue
+                        }
+                        v := c.FindFileShareValue(key, vol)
+                        if !strings.EqualFold(m[key][0], v) {
+                                flag = false
+                                break
+                        }
+                }
+                if flag {
+                        vols = append(vols, vol)
+                }
+        }
+        return vols
+
+}
+
+func (c *Client) SortFileShares(shares []*model.FileShareSpec, p *Parameter) []*model.FileShareSpec {
```

Comment:
We can plan unit tests to cover all these functions! If these are already covered during your POST and GET tests then probably a new PR can be raised for unit tests!